src/components/structures/RoomView.tsx

```diff
@@ -1208,12 +1211,30 @@ export class RoomView extends React.Component<IRoomProps, IRoomState> {
             case Action.EditEvent: {
                 // Quit early if we're trying to edit events in wrong rendering context
                 if (payload.timelineRenderingType !== this.state.timelineRenderingType) return;
+                if (payload.event && payload.event.getRoomId() !== this.state.roomId) {
+                    // If the event is in a different room (e.g. all rooms search), we need to view that room first
+                    dis.dispatch<ViewRoomPayload>({
+                        action: Action.ViewRoom,
+                        room_id: payload.event?.getRoomId(),
+                        metricsTrigger: undefined,
+                        deferred_action: payload,
+                    });
+                    return;
+                }
+
                 const editState = payload.event ? new EditorStateTransfer(payload.event) : undefined;
-                this.setState({ editState }, () => {
-                    if (payload.event) {
-                        this.messagePanel?.scrollToEventIfNeeded(payload.event.getId());
-                    }
-                });
+                this.setState(
+                    {
+                        editState,
+                        // Close any ongoing search as the RoomSearchView doesn't pass editState and thus won't render the edit composer
```

Comment:
pass it to where?

I think I would phrase this as something like:

> If a search is active (implying that the "edit" button has been pressed on one of the events in the search result), we need to close that search, because `RoomSearchView` doesn't handle editing and won't render the composer.